### PR TITLE
[SPARK-7605] [MLlib] [PySpark] Python API for ElementwiseProduct

### PR DESCRIPTION
**Pull request by MechCoder**
Python API for org.apache.spark.mllib.feature.ElementwiseProduct

### CODE REVIEW

**Comment by MechCoder** [comment]
ping @mengxr @jkbradley

**Comment by MechCoder** [comment]
ping @davies Would you be able to have a look at this?

File: python/pyspark/mllib/feature.py [hunk 1]
```diff
@@ -525,6 +526,41 @@ def fit(self, data):
...
+    >>> eprod.transform(a)
+    DenseVector([2.0, 2.0, 9.0])
+    >>> b = Vectors.dense([9.0, 3.0, 4.0])
+    >>> rdd = sc.parallelize([a, b])
+    >>> eprod.transform(rdd).collect()
+    [DenseVector([2.0, 2.0, 9.0]), DenseVector([9.0, 6.0, 12.0])]
+    """
+    def __init__(self, vector):
+        if not isinstance(vector, Vector):
+            raise ValueError(
```
Comment (davies) [inline python/pyspark/mllib/feature.py]: Should this be TypeError?

File: python/pyspark/mllib/feature.py [hunk 2]
```diff
@@ -525,6 +526,41 @@ def fit(self, data):
...
+    >>> a = Vectors.dense([2.0, 1.0, 3.0])
+    >>> eprod.transform(a)
+    DenseVector([2.0, 2.0, 9.0])
+    >>> b = Vectors.dense([9.0, 3.0, 4.0])
+    >>> rdd = sc.parallelize([a, b])
+    >>> eprod.transform(rdd).collect()
+    [DenseVector([2.0, 2.0, 9.0]), DenseVector([9.0, 6.0, 12.0])]
+    """
+    def __init__(self, vector):
+        if not isinstance(vector, Vector):
```
Comment (davies) [inline python/pyspark/mllib/feature.py]: It will be good to support list and np.array

File: python/pyspark/mllib/feature.py (hunk 1 above)
Comment (MechCoder) [inline python/pyspark/mllib/feature.py]: Hmm. on Second thoughts, I should maybe just write

self.vector = _convert_to_vector(vector)

WDYT?

**Comment by MechCoder** [comment]
@davies I fixed your comments and replaced vector with scalingVector for consistency

File: python/pyspark/mllib/feature.py (hunk 1 above)
Comment (davies) [inline python/pyspark/mllib/feature.py]: Sounds good.

**Comment by davies** [comment]
LGTM, merging into master!

**Comment by MechCoder** [comment]
@davies Thanks. Do you want me to look at any particular PR if it needs reviews?

**Comment by davies** [comment]
Any comments on any PR or JIRA are welcome, thanks!

File: docs/mllib-feature-extraction.md [hunk 3]
```diff
@@ -527,6 +527,28 @@ val transformedData2 = parsedData.map(x => transformer.transform(x))
 
 {% endhighlight %}
 </div>
+
+<div data-lang="python">
+{% highlight python %}
+from pyspark import SparkContext
```
Comment (jkbradley) [inline docs/mllib-feature-extraction.md]: No need to set up SparkContext for an example here.  I prefer to be able to copy and paste code into the pyspark shell for testing.

File: docs/mllib-feature-extraction.md [hunk 4]
```diff
@@ -527,6 +527,28 @@ val transformedData2 = parsedData.map(x => transformer.transform(x))
...
+# Load and parse the data
+sc = SparkContext()
+data = sc.textFile("data/mllib/kmeans_data.txt")
+parsedData = data.map(lambda x: [float(t) for t in x.split(" ")])
+
+# Create weight vector.
+transformingVector = Vectors.dense([0.0, 1.0, 2.0])
+transformer = ElementwiseProduct(transformingVector)
+
+# Batch transform.
```
Comment (jkbradley) [inline docs/mllib-feature-extraction.md]: Batch and single-row transform.

**Comment by jkbradley** [comment]
Oh, I see this was merged already.  @MechCoder I'm updating some docs, so I'll include those comments in my next patch.  Thanks for the PR!

**Pull request closed without merging**